### PR TITLE
DATAUP-497: Moving all bus message sending to the TestUtils file

### PR DESCRIPTION
**Pull request by ialarmedalien**
# Description of PR purpose/changes

Centralising all bus message sends in the TestUtils files for ease of manipulation.

# Jira Ticket / Issue #

Related Jira ticket: https://kbase-jira.atlassian.net/browse/DATAUP-497
- [x] Added the Jira Ticket to the title of the PR (e.g. `DATAUP-69 Adds a PR template`)

# Testing Instructions
* Details for how to test the PR:
- [x] Tests pass locally and in GitHub Actions
- [ ] Changes available by spinning up a local narrative and navigating to _X_ to see _Y_

# Dev Checklist:

- [x] My code follows the guidelines at https://sites.google.com/lbl.gov/trussresources/home?authuser=0
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] (JavaScript) I have run Prettier and ESLint on changed code manually or with a git precommit hook

### CODE REVIEW

File: kbase-extension/static/kbase/js/common/jobManager.js [hunk 1]
```diff
@@ -276,18 +276,18 @@ define([
             }
 
             channelList
-                .filter((channel) => {
-                    return channel && channel.length > 0 ? 1 : 0;
+                .filter((channelId) => {
+                    return channelId && channelId.length > 0 ? 1 : 0;
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/common/jobManager.js:280]: minor nomenclature change

File: test/unit/spec/common/cellComponents/tabs/jobStatusTable-Spec.js [hunk 2]
```diff
@@ -1115,7 +1017,18 @@ define([
                             this.input.meta.retryTarget = this.job.job_id;
                             spyOn(this.jobManager, 'updateModel').and.callThrough();
                             await TestUtil.waitForElementChange(this.row, () => {
-                                send_STATUS(this);
+                                // TODO: replace with a genuine retry instead --
+                                // the jobState is for a job with a different ID!
```
Comment (ialarmedalien) [inline test/unit/spec/common/cellComponents/tabs/jobStatusTable-Spec.js:1021]: hadn't realised I had done this bit of hackery! 😱

File: test/unit/spec/common/cellComponents/tabs/jobStatusTable-Spec.js [hunk 3]
```diff
@@ -1127,36 +1040,6 @@ define([
                     });
                 });
 
-                describe('invalid', () => {
```
Comment (ialarmedalien) [inline test/unit/spec/common/cellComponents/tabs/jobStatusTable-Spec.js:1130]: these messages won't make it past the job manager, so there isn't much point in keeping these tests in

File: test/unit/spec/common/cellComponents/tabs/jobStatusTable-Spec.js [hunk 4]
```diff
@@ -1534,39 +1424,6 @@ define([
...
```
Comment (ialarmedalien) [inline test/unit/spec/common/cellComponents/tabs/jobStatusTable-Spec.js:1537]: ditto here - more invalid messages

File: test/unit/spec/nbextensions/appCell2/widgets/appCellWidget-spec.js [hunk 5]
```diff
@@ -223,6 +168,7 @@ define([
             bus: ctx.bus,
             cell: mockCell,
             devMode: true,
+            saveMaxFrequency: 1,
```
Comment (ialarmedalien) [inline test/unit/spec/nbextensions/appCell2/widgets/appCellWidget-spec.js:171]: Prevent the save delay from slowing down the tests

File: test/unit/spec/nbextensions/appCell2/widgets/appCellWidget-spec.js [hunk 6]
```diff
@@ -621,15 +580,7 @@ define([
...
                         await TestUtil.waitForElementChange(
-                            this.kbaseNode.querySelector('[data-button="jobStatus"]'),
-                            () => {
-                                // send a status update
-                                send_STATUS({
-                                    bus: this.bus,
-                                    jobId,
-                                    jobState,
-                                });
-                            }
```
Comment (ialarmedalien) [inline test/unit/spec/nbextensions/appCell2/widgets/appCellWidget-spec.js:632]: move this so it's done in response to the `START_UPDATE` message, as would happen IRL

File: test/unit/spec/util/jobLogViewerSpec.js [hunk 7]
```diff
@@ -30,52 +30,14 @@ define([
 
     function createLogViewer(context, showHistory = false, logPollInterval = null) {
         context.node = document.createElement('div');
-        context.runtimeBus = Runtime.make().bus();
```
Comment (ialarmedalien) [inline test/unit/spec/util/jobLogViewerSpec.js:33]: changed to `bus` throughout this file

**Review by briehl — APPROVED** [review]

**Pull request merged**